lib/trino-parquet/src/test/java/io/trino/parquet/reader/BenchmarkBinaryColumnReader.java

```diff
@@ -0,0 +1,230 @@
+    public enum FieldType
+    {
+        UNBOUNDED(range -> VARBINARY, BenchmarkBinaryColumnReader::randomData),
+        VARCHAR_ASCII_BOUND_EXACT(range -> VarcharType.createVarcharType(max(1, range.to)), BenchmarkBinaryColumnReader::randomAsciiData),
+        CHAR_ASCII_BOUND_HALF(range -> CharType.createCharType(max(1, range.to / 2)), BenchmarkBinaryColumnReader::randomAsciiData),
+        CHAR_BOUND_HALF_PADDING_SOMETIMES(range -> CharType.createCharType(max(1, range.to / 2)), (length, range) -> randomAsciiDataWithPadding(length, range, .01)),
+        /**/;
+
+        private final Function<Range, Type> type;
+        private final BiFunction<Integer, Range, byte[][]> dataGenerator;
+
+        public Type getType(Range length)
+        {
+            return type.apply(length);
+        }
+
+        public byte[][] generateData(int size, Range positionLength)
+        {
+            return dataGenerator.apply(size, positionLength);
+        }
+
+        FieldType(Function<Range, Type> type, BiFunction<Integer, Range, byte[][]> dataGenerator)
+        {
+            this.type = requireNonNull(type, "type is null");
+            this.dataGenerator = requireNonNull(dataGenerator, "dataGenerator is null");
+        }
+    }
+
+    public enum PositionLength
+    {
+        VARIABLE_0_100(0, 100),
+        VARIABLE_0_1000(0, 1000),
+        FIXED_10(10, 10),
+        FIXED_100(100, 100),
+        /**/;
+
+        private final Range range;
+
+        public Range getRange()
+        {
+            return range;
+        }
+
+        PositionLength(int from, int to)
+        {
+            this.range = new Range(from, to);
+        }
+    }
+
+    private static final class Range
+    {
+        public int from;
+        public int to;
+
+        public Range(int from, int to)
+        {
+            this.from = from;
+            this.to = to;
+        }
+    }
+
+    private static byte[][] randomData(int size, Range positionLength)
+    {
+        Random random = new Random(Objects.hash(size, positionLength.from, positionLength.to));
+        byte[][] data = new byte[size][];
+        for (int i = 0; i < size; i++) {
+            int length = random.nextInt(positionLength.to - positionLength.from + 1) + positionLength.from;
+            byte[] value = new byte[length];
+            random.nextBytes(value);
+            data[i] = value;
+        }
+
+        return data;
+    }
+
+    private static byte[][] randomAsciiData(int size, Range positionLength)
+    {
+        Random random = new Random(Objects.hash(size, positionLength.from, positionLength.to));
+        byte[][] data = new byte[size][];
+        for (int i = 0; i < size; i++) {
+            int length = random.nextInt(positionLength.to - positionLength.from + 1) + positionLength.from;
+            byte[] value = new byte[length];
+            random.nextBytes(value);
+            for (int j = 0; j < length; j++) {
+                value[j] = (byte) (value[j] >>> 1);
+            }
```

Comment:
This is an awkward way to do it. You can just generate a random value between 0 and 127 for each slot:

```java
for (int j = 0; j < length; j++) {
    value[j] = (byte) random.nextInt(128);
}
```